### DISCUSSION TITLE
Scalar API documentation is not working with .Net 9

### DISCUSSION THREAD

**Discussion by rushjam** [scalar/scalar#3659] [Q&A]
I have created .net 9 preview web api using default starter template with Scalar.AspNetCore (1.2.24).

But I don't see any endpoints being rendered (In my case WeatherForecase that comes with default template). I just see blank page. How can I fix this?

Open API Spec ->  [openapi.json](https://github.com/user-attachments/files/17550420/openapi.json)

![image](https://github.com/user-attachments/assets/832c9ce7-9016-4ff0-b1fa-0e0c558bccb3)

**Comment by amritk** [comment]
Hello, without seeing your spec I can't be 100% sure but it might be fixed in this PR

https://github.com/scalar/scalar/pull/3658

**Reply by rushjam** [reply]
Thanks @amritk. I have attached OpenAPI Spec in the original question.

**Reply by amritk** [reply]
hmm that spec seems to work fine. @xC0dex any ideas?

**Reply by xC0dex** [reply]
This behavior occurs when the requested document was not found under the given path. So I would assume that there is no document available under `/openapi/{documentName}.json`. `{documentName}` is the name of the OpenAPI document (By default `v1`). @rushjam please check out the [docs](https://github.com/scalar/scalar/blob/main/documentation/integrations/dotnet.md). There are guides on how to set up Scalar.

If It's still not working, I would appreciate it if you could share the content of the `Program.cs`.

**Reply by rushjam** [reply]
@xC0dex, I can able to open OpenAPI Document under `/openapi/v1`. For the reference, I have shared content of the `program.cs` as well. Thanks.

![image](https://github.com/user-attachments/assets/fe41c219-38e1-4941-9ead-1ef4f4d3ebf1)

```cs
using Scalar.AspNetCore;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.

builder.Services.AddControllers();
// Learn more about configuring OpenAPI at https://aka.ms/aspnet/openapi
builder.Services.AddOpenApi();

var app = builder.Build();

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
{
    app.MapOpenApi();
    app.MapScalarApiReference(options =>
    {
        options.WithTitle("Demo Scalar 9.0");
    });
}

app.UseHttpsRedirection();

app.UseAuthorization();

app.MapControllers();

app.Run();
```

**Reply by xC0dex** [reply]
This is very strange. It should work out of the box. Does the log provide any useful information?
I wondered why `app.UseAuthorization()` is working without calling `services.AddAuthorization()`. Unfortunately, I cannot reproduce it 😕.

**Comment by xC0dex** [comment]
Hey @rushjam,

Did you update the OpenAPI document name to `iv1`? Because this is what I can see in your screenshot. The default document name is `v1`, so it's maybe just a typo in your URL.

**Reply by rushjam** [reply]
Yes @xC0dex, you're correct. That was a typo. However, I still see the same blank page no matter what I type, even with typing (v1).

**Comment by MonokeKumu** [comment]
Hey
By adding this option in scalar mapping, I got the error fixed.
options.WithOpenApiRoutePattern("/openapi/v1.json");
